bitcoin/src/blockdata/locktime/relative.rs

```diff
@@ -270,47 +276,58 @@ impl fmt::Display for Time {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
 }
 
-/// Errors related to relative lock times.
+/// Input time in seconds was too large to be encoded to a 16 bit 512 second interval.
 #[derive(Debug, Clone, PartialEq, Eq)]
-#[non_exhaustive]
-pub enum Error {
-    /// Input time in seconds was too large to be encoded to a 16 bit 512 second interval.
-    IntegerOverflow(u32),
-    /// Tried to satisfy a lock-by-blocktime lock using a height value.
-    IncompatibleHeight(LockTime, Height),
-    /// Tried to satisfy a lock-by-blockheight lock using a time value.
-    IncompatibleTime(LockTime, Time),
+pub struct IntegerOverflowError {
+    /// Time value in seconds that overflowed.
+    pub seconds: u32
```

Comment:
We can just hide the representation leaving the option to add it later. (And have a getter today.)

Reply:
We can do this but it seems excessive for an error type:
```rust
/// Input time in seconds was too large to be encoded to a 16 bit 512 second interval.
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct TimeOverflowError {
    /// Time value in seconds that overflowed.
    seconds: u32                // We maintain an invariant that this value does indeed overflow.
}

impl TimeOverflowError {
    pub fn new(seconds: u32) -> Self {
        let overflows = u16::try_from((seconds + 511) / 512).is_ok();
        assert!(overflows);
        TimeOverflowError { seconds }
    }
}
```
I personally think the error is fine as it is. We are not here to protect every programmer from every single programming mistake they might make.

Reply:
@tcharding the cool thing is that we don't need `new` because users should never be directly constructing our error types.

Reply:
@tcharding the cool thing is that we don't need `new` because users should never be directly constructing our error types.

And internally of course we can just directly construct them.

Reply:
So just leave this PR as is, right? (FTR the `pub` was an accident, and also for the record its Monday morning and I've lost where we got upto on policy of hider errors - my current understanding is public fields, use `non_exhaustive`, optional private constructor)

Reply:
I believe this thread is resolved and requires no action.

Reply:
@tcharding I don't think so. I still see the field as `pub`, you didn't address @apoelstra's comment and I agree with his comment.

Reply:
I made the field `pub(crate)` and added a comment so it doesn't get un-privated.

Reply:
Note also that the constructor has been removed - I believe this thread is resolved.

Reply:
I am going to just disable the "resolve conversation" rule.